cmd/snap-update-ns/update.go

```diff
@@ -0,0 +1,44 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"github.com/snapcore/snapd/osutil"
+)
+
+// MountProfileUpdate abstracts the operations around mount namespace updates.
+type MountProfileUpdate interface {
```

Comment:
this interface is a bit odd in terms of internal/external responsibilities, let's see how it evolves over the next PRs

Reply:
to clarify what I mean by odd:  there's too much passing in to it things that come out of it as well

Reply:
We discussed this earlier today and I think this is now resolved.